src/trie/map.rs

```diff
@@ -187,8 +187,8 @@ impl<T> TrieMap<T> {
     /// assert_eq!(vec, vec!["c", "b"]);
     /// ```
     #[inline]
-    pub fn each_reverse<'a>(&'a self, f: |&uint, &'a T| -> bool) -> bool {
-        self.root.each_reverse(f)
+    pub fn each_reverse<F>(&self, mut f: F) -> bool where F: FnMut(&uint, &T) -> bool {
```

Comment:
Thanks for pointing that out! I've restored the explicit lifetimes and changed `trie_map::test::test_each_reverse` to ensure that your example works.